### PR TITLE
[Keyboard] Add new keyboards by Lelelab

### PR DESCRIPTION
**Pull request by andywith**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/lelelab/lelepad/keymaps/lefty/keymap.c [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+#include "quantum.h"
+#include "matrix.h"
```
Comment (drashna) [inline keyboards/lelelab/lelepad/keymaps/lefty/keymap.c:18]:
```suggestion
```

File: keyboards/lelelab/lelepad/keymaps/lefty/keymap.c [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        oled_write_raw_P(lele_logo,128*4);
+    }
+}
+
+void oled_task_user(void) {
+    icon_draw();
+}
+
+void keyboard_post_init_user(void) {
+    oled_init(0);
```
Comment (drashna) [inline keyboards/lelelab/lelepad/keymaps/lefty/keymap.c:61]: This shouldn't be needed. 
```suggestion
```

File: keyboards/lelelab/lelepad/keymaps/lefty/keymap.c [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+uint8_t draw_once_flag = 0;
+
+static void icon_draw(void) {
+    if (draw_once_flag) {
+        draw_once_flag = 0;
+        oled_write_raw_P(lele_logo,128*4);
+    }
+}
+
+void oled_task_user(void) {
```
Comment (drashna) [inline keyboards/lelelab/lelepad/keymaps/lefty/keymap.c:56]: This should be moved to the keyboard's c file.

File: keyboards/lelelab/lelepad/keymaps/righty/keymap.c [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+#include "quantum.h"
+#include "matrix.h"
```
Comment (drashna) [inline keyboards/lelelab/lelepad/keymaps/righty/keymap.c:18]:
```suggestion
```

File: keyboards/lelelab/lelepad/lelepad.c [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+
+void matrix_init_kb(void) {
+    SLED_INIT();
+    SLED_OFF();
+}
+
+bool led_update_kb(led_t led_state) {
+    return led_update_user(led_state);
+}
```
Comment (drashna) [inline keyboards/lelelab/lelepad/lelepad.c:26]:
```suggestion
```

File: keyboards/lelelab/lelepad/lelepad.h [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    {k41,k42,k43,k44},\
+    {k51,k52,k53,k54}\
+}
+
+
+// status LED
+#define SLED_INIT()   DDRD |= (1<<4)
+#define SLED_ON()     PORTD &= ~(1<<4)
+#define SLED_OFF()    PORTD |= (1<<4)
+#define SLED_TOGGLE()   PORTD ^= (1<<4)
```
Comment (drashna) [inline keyboards/lelelab/lelepad/lelepad.h:41]: Please convert these to GPIO commands.

https://docs.qmk.fm/#/internals_gpio_control

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/lelelab/lele60/info.json [hunk 7]
```diff
@@ -0,0 +1,82 @@
+{
+    "keyboard_name": "LelePad",
+    "maintainer": "LeleLab",
+    "url": "https://www.zfrontier.com/app/mch/oNZEw6bvZ3Go",
+    "layouts": {
+        "LAYOUT_all": {
```
Comment (zvecr) [inline keyboards/lelelab/lele60/info.json:6]:
```suggestion
        "LAYOUT": {
```

File: keyboards/lelelab/lelepad/info.json [hunk 8]
```diff
@@ -0,0 +1,31 @@
+{
+    "keyboard_name": "LelePad",
+    "maintainer": "LeleLab",
+    "url": "https://en.zfrontier.com/products/in-stock-lelepad-custom-numpad-macropad-keyboard-kit",
+    "layouts": {
+        "LAYOUT_all": {
```
Comment (zvecr) [inline keyboards/lelelab/lelepad/info.json:6]:
```suggestion
        "LAYOUT": {
```

File: keyboards/lelelab/lelepad/rules.mk [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+VIA_ENABLE = yes
```
Comment (zvecr) [inline keyboards/lelelab/lelepad/rules.mk:24]: Should only be enabled at the keymap level
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]
Also, `keyboards/lelelab/lelepad` has no `default` keymap.

**Comment by andywith** [comment]
Hello 
How do I do a Pull Request again?

**Comment by peepeetee** [comment]
> 
> 
> Hello How do I do a Pull Request again?

image
https://docs.qmk.fm/#/newbs_git_using_your_master_branch?id=making-changes

**Comment by peepeetee** [comment]
Follow up on my previous comment, you are in a position where you should um
you should make a new branch that is your current master, and make that the source for this PR.

After that you can fix your master by pulling upstream master, and then branch from your master

**Comment by peepeetee** [comment]
您看起来需要帮助，我来帮您git？
给我个qq号？

For the records here I am proposing to help them with git and asking for an IM username

**Comment by andywith** [comment]
reopen #15105   , close  pr #15165

File: keyboards/lelelab/lele60/readme.md [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+* Hardware Supported: Lele60 PCB, which is GH60 compatible.
+* Hardware Availability: keyboard is distributed by zFrontier at [zFrontier.com](https://www.zfrontier.com/app/mch/oNZEw6bvZ3Go) in China, and [zFrontier.en](https://en.zfrontier.com/products/lele60-crystalline) globally.
+
+Make example for this keyboard (after setting up your build environment):
+
+    make lele60:default
+
+Flashing example for this keyboard:
+
+    make lele60:default:flash
```
Comment (peepeetee) [inline keyboards/lelelab/lele60/readme.md:17]:
```suggestion
    make lelelab/lele60:default

Flashing example for this keyboard:

    make lelelab/lele60:default:flash
```

File: keyboards/lelelab/lelepad/readme.md [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+* Hardware Supported: LelePad PCB
+* Hardware Availability: LelePad is distributed by zFrontier at [zFrontier.com](https://www.zfrontier.com/app/mch/lAO36WKQ8XXB) in China, and [zFrontier.en](https://en.zfrontier.com/products/in-stock-lelepad-custom-numpad-macropad-keyboard-kit) globally.
+
+Make example for this keyboard (after setting up your build environment):
+
+    make lelepad:righty
+
+Flashing example for this keyboard:
+
+    make lelepad:righty:flash
```
Comment (peepeetee) [inline keyboards/lelelab/lelepad/readme.md:17]:
```suggestion
    make lelelab/lelepad:righty

Flashing example for this keyboard:

    make lelelab/lelepad:righty:flash
```

**Review by peepeetee — CHANGES_REQUESTED** [review]

**Pull request closed without merging**